lib/valvat/lookup.rb

```diff
@@ -28,15 +31,18 @@ def valid?
     end
 
     def response
-      @response ||= Request.new(@vat, @options).perform
+      @response ||= case @vat.vat_country_code
+      when 'GB' then HMRC::Request.new(@vat, @options).perform
+      else VIES::Request.new(@vat, @options).perform
+      end
```

Comment:
This is where the main switching is done, for a "pluggable" version this code would need to be a bit more advanced but for now this is enough.